### PR TITLE
#1642 Correct investment amount calculation

### PR DESCRIPTION
**Pull request by desfero**

### CODE REVIEW

File: app/components/dashboard/eto-list/EtoList.tsx [hunk 1]
```diff
@@ -28,39 +25,9 @@ const EtoListComponent: React.SFC<IProps> = ({ etos }) => (
     </Col>
     {etos &&
       etos.map(eto => (
-        <Col xs={12} key={eto.etoId}>
+        <Col xs={12} key={eto.previewCode}>
           <div className="mb-3">
-            <EtoOverviewStatus
-              preMoneyValuationEur={eto.preMoneyValuationEur}
```
Comment (desfero) [inline app/components/dashboard/eto-list/EtoList.tsx]: @Moejoe90 @rudolfix i passed whole eto as it's starting to be complicated and i saw that a couple of props where duplicated (for e.g. newSharesGenerated and newSharesToIssue)

File: app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx [hunk 2]
```diff
@@ -74,59 +53,61 @@ const PoweredByNeufund = () => {
...
+  eto,
+  isAuthorized,
+  isEligibleToPreEto,
+}: IExternalProps & IStateProps & IWithIsEligibleToPreEto) => {
   // It's possible for contract to be undefined if eto is not on chain yet
-  const timedState = props.contract ? props.contract.timedState : ETOStateOnChain.Setup;
+  const timedState = eto.contract ? eto.contract.timedState : ETOStateOnChain.Setup;
 
   switch (timedState) {
     case ETOStateOnChain.Setup: {
```
[the remark this answers is not in the document]
Comment (desfero) [reply app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx]: Good point, let's leave it until component is stable as we still need to do some changes for iframe version that may affect how components are rendered

File: app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx [hunk 3]
```diff
@@ -74,59 +53,61 @@ const PoweredByNeufund = () => {
...
+              maxPledge={eto.maxTicketEur || 0}
+              minPledge={eto.minTicketEur || 0}
+              etoId={eto.etoId}
+              investorsLimit={eto.maxPledges || 0}
             />
           );
         } else if (
-          props.contract &&
-          props.contract.startOfStates[ETOStateOnChain.Whitelist]! > new Date()
+          eto.contract &&
```
[the remark this answers is not in the document]
Comment (desfero) [reply app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx]: change to `isWaitingForWhitelistToStart`

File: app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx [hunk 4]
```diff
@@ -74,59 +53,61 @@ const PoweredByNeufund = () => {
...
             <CounterWidget
               endDate={
-                props.contract.startOfStates[
-                  props.isEligibleToPreEto ? ETOStateOnChain.Whitelist : ETOStateOnChain.Public
+                eto.contract.startOfStates[
+                  isEligibleToPreEto ? ETOStateOnChain.Whitelist : ETOStateOnChain.Public
                 ]!
               }
-              stage={props.isEligibleToPreEto ? "PRE-ETO" : "ETO"}
+              stage={isEligibleToPreEto ? "PRE-ETO" : "ETO"}
```
[the remark this answers is not in the document]
Comment (desfero) [reply app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx]: fixed

File: app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx [hunk 5]
```diff
@@ -74,59 +53,61 @@ const PoweredByNeufund = () => {
...
     case ETOStateOnChain.Whitelist: {
-      if (props.isEligibleToPreEto) {
+      if (isEligibleToPreEto) {
         return (
           <InvestWidget
-            raisedTokens={props.contract!.totalInvestment.totalTokensInt.toNumber()}
-            investorsBacked={props.contract!.totalInvestment.totalInvestors.toNumber()}
-            tokensGoal={(props.newSharesToIssue || 1) * (props.equityTokensPerShare || 1)}
-            etoId={props.etoId}
+            raisedTokens={eto.contract!.totalInvestment.totalTokensInt.toNumber()}
```
[the remark this answers is not in the document]
Comment (desfero) [reply app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx]: so it's quite complicated and i already thought about a way to improve it somehow as whether the property is available or not depends on the state of the eto and when this code is executed eto is always in ON_CHAIN state. Maybe we can use typescript unit types but still need to research how to do it the best way :(

File: app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx [hunk 6]
```diff
@@ -195,9 +181,9 @@ const EtoOverviewStatusLayout: React.SFC<
 
         <div className={styles.tagsWrapper}>
           <TagsWidget
-            etoId={props.etoId}
-            termSheet={props.termSheet}
-            prospectusApproved={props.prospectusApproved}
+            etoId={eto.etoId}
+            termSheet={documentsByType["termsheet_template"]}
```
[the remark this answers is not in the document]
Comment (desfero) [reply app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx]: good catch, fixed

File: app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Moejoe90) [reply app/components/eto/overview/EtoOverviewStatus/EtoOverviewStatus.tsx]: :+1:

File: app/components/shared/ETOState.tsx [hunk 7]
```diff
@@ -29,7 +29,7 @@ interface IStateProps {
   eto: TEtoWithCompanyAndContract;
 }
 
-const statusToName: Record<EtoState | ETOStateOnChain, React.ReactNode> = {
+export const statusToName: Record<EtoState | ETOStateOnChain, JSX.Element> = {
```
Comment (Moejoe90) [inline app/components/shared/ETOState.tsx]: what's the difference between `JSX.Element` and `React.ReactNode`?

**Pull request merged**